inc/editor/configuration-loader/configuration-loader.php

```diff
@@ -208,7 +208,7 @@ private static function register_selector( string $block_title, string $type, Qu
 		);
 	}
 
-	public static function register_query( string $block_title, QueryContext $query ): void {
+	public static function register_query( string $block_title, QueryContextInterface $query ): void {
```

Comment:
This is the practical why--this and increasing mental model accuracies. I don't see any blockers to the core code handling anything that conforms to what we have in the base context there.